### PR TITLE
Expose ingest storage config doc and changelog

### PR DESCRIPTION
**Pull request by pracucci**
#### What this PR does

The Kafka-based ingest storage is still experimental, but we're moving closer to start running it in few production clusters at Grafana Labs. I think it's the right time to include it in the auto-generated config and, from now on, to mention changes to the CHANGELOG.

I've added the CHANGELOG entry under `main / unreleased` branch, even if it's not technically correct, because all changes are included in `v2.13.0-rc.0`. I'm open to feedback on how to handle it. The reason why I've written it under `main / unreleased` is because we haven't publicly mentioned it yet, but will be mentioned as experimental feature in the next release (so 2.14).

#### Which issue(s) this PR fixes or relates to

N/A

#### Checklist

- [ ] Tests updated.
- [ ] Documentation added.
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [ ] [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

File: cmd/mimir/help-all.txt.tmpl [hunk 1]
```diff
@@ -1305,6 +1305,46 @@ Usage of ./cmd/mimir/mimir:
...
     	HTTP URL path under which the Prometheus api will be served. (default "/prometheus")
+  -ingest-storage.enabled
+    	True to enable the ingestion via object storage.
+  -ingest-storage.ingestion-partition-tenant-shard-size int
+    	[experimental] The number of partitions a tenant's data should be sharded to when using the ingest storage. Tenants are sharded across partitions using shuffle-sharding. 0 disables shuffle sharding and tenant is sharded across all partitions.
+  -ingest-storage.kafka.address string
+    	The Kafka backend address.
+  -ingest-storage.kafka.auto-create-topic-default-partitions int
+    	When auto-creation of Kafka topic is enabled and this value is positive, Kafka's num.partitions configuration option is set on Kafka brokers with this value when Mimir component that uses Kafka starts. This configuration option specifies the default number of partitions that Kafka broker will use for auto-created topics. Note that this is Kafka-cluster wide setting, and applies to any auto-created topic. If setting of num.partitions fails, Mimir will proceed anyway, but auto-created topic may have incorrect number of partitions.
```
Comment (tacole02) [inline cmd/mimir/help-all.txt.tmpl:1315]:
```suggestion
    	When auto-creation of Kafka topic is enabled and this value is positive, Kafka's num.partitions configuration option is set on Kafka brokers with this value when Mimir component that uses Kafka starts. This configuration option specifies the default number of partitions that the Kafka broker uses for auto-created topics. Note that this is a Kafka-cluster wide setting, and applies to any auto-created topic. If the setting of num.partitions fails, Mimir proceeds, but the auto-created topic could have an incorrect number of partitions.
```

File: cmd/mimir/help-all.txt.tmpl [hunk 2]
```diff
@@ -1305,6 +1305,46 @@ Usage of ./cmd/mimir/mimir:
...
+  -ingest-storage.kafka.auto-create-topic-enabled
+    	Enable auto-creation of Kafka topic if it doesn't exist. (default true)
+  -ingest-storage.kafka.client-id string
+    	The Kafka client ID.
+  -ingest-storage.kafka.consume-from-position-at-startup string
+    	From which position to start consuming the partition at startup. Supported options: last-offset, start, end, timestamp. (default "last-offset")
+  -ingest-storage.kafka.consume-from-timestamp-at-startup int
+    	Milliseconds timestamp after which the consumption of the partition starts at startup. Only applies when consume-from-position-at-startup is timestamp
+  -ingest-storage.kafka.consumer-group string
+    	The consumer group used by the consumer to track the last consumed offset. The consumer group must be different for each ingester. If the configured consumer group contains the '<partition>' placeholder, it will be replaced with the actual partition ID owned by the ingester. When empty (recommended), Mimir will use the ingester instance ID to guarantee uniqueness.
```
Comment (tacole02) [inline cmd/mimir/help-all.txt.tmpl:1325]:
```suggestion
    	The consumer group used by the consumer to track the last consumed offset. The consumer group must be different for each ingester. If the configured consumer group contains the '<partition>' placeholder, it is replaced with the actual partition ID owned by the ingester. When empty (recommended), Mimir will use the ingester instance ID to guarantee uniqueness.
```

File: cmd/mimir/help-all.txt.tmpl [hunk 3]
```diff
@@ -1305,6 +1305,46 @@ Usage of ./cmd/mimir/mimir:
...
+    	How long to retry a failed request to get the last produced offset. (default 10s)
+  -ingest-storage.kafka.max-consumer-lag-at-startup duration
+    	The maximum tolerated lag before a consumer is considered to have caught up reading from a partition at startup, becomes ACTIVE in the hash ring and passes the readiness check. Set 0 to disable waiting for maximum consumer lag being honored at startup. (default 15s)
+  -ingest-storage.kafka.producer-max-record-size-bytes int
+    	The maximum size of a Kafka record data that should be generated by the producer. An incoming write request bigger than this size is split into multiple Kafka records. We strongly recommend to not change this setting unless for testing purposes. (default 15983616)
+  -ingest-storage.kafka.topic string
+    	The Kafka topic name.
+  -ingest-storage.kafka.write-clients int
+    	The number of Kafka clients used by producers. When the configured number of clients is greater than 1, partitions are sharded among Kafka clients. An higher number of clients may provide higher write throughput at the cost of additional Metadata requests pressure to Kafka. (default 1)
```
Comment (tacole02) [inline cmd/mimir/help-all.txt.tmpl:1341]:
```suggestion
    	The number of Kafka clients used by producers. When the configured number of clients is greater than 1, partitions are sharded among Kafka clients. A higher number of clients could provide higher write throughput at the cost of additional Metadata requests pressure to Kafka. (default 1)
```

File: cmd/mimir/help-all.txt.tmpl [hunk 4]
```diff
@@ -1305,6 +1305,46 @@ Usage of ./cmd/mimir/mimir:
...
+  -ingest-storage.kafka.write-timeout duration
+    	How long to wait for an incoming write request to be successfully committed to the Kafka backend. (default 10s)
+  -ingest-storage.migration.distributor-send-to-ingesters-enabled
+    	When both this option and ingest storage is enabled, distributors will both write to Kafka and ingesters. A write request will be considered successful only when written to both backends.
```
Comment (tacole02) [inline cmd/mimir/help-all.txt.tmpl:1345]:
```suggestion
    	When both this option and ingest storage is enabled, distributors write to both Kafka and ingesters. A write request is considered successful only when written to both backends.
```

File: cmd/mimir/help-all.txt.tmpl [hunk 5]
```diff
@@ -1407,6 +1447,64 @@ Usage of ./cmd/mimir/mimir:
...
+  -ingester.partition-ring.consul.consistent-reads
+    	Enable consistent reads to Consul.
+  -ingester.partition-ring.consul.hostname string
+    	Hostname and port of Consul. (default "localhost:8500")
+  -ingester.partition-ring.consul.watch-burst-size int
+    	Burst size used in rate limit. Values less than 1 are treated as 1. (default 1)
+  -ingester.partition-ring.consul.watch-rate-limit float
+    	Rate limit when watching key or prefix in Consul, in requests per second. 0 disables the rate limit. (default 1)
+  -ingester.partition-ring.delete-inactive-partition-after duration
+    	How long to wait before an INACTIVE partition is eligible for deletion. The partition will be deleted only if it has been in INACTIVE state for at least the configured duration and it has no owners registered. A value of 0 disables partitions deletion. (default 13h0m0s)
```
Comment (tacole02) [inline cmd/mimir/help-all.txt.tmpl:1465]:
```suggestion
    	How long to wait before an INACTIVE partition is eligible for deletion. The partition is deleted only if it has been in INACTIVE state for at least the configured duration and has no owners registered. A value of 0 disables partitions deletion. (default 13h0m0s)
```

File: cmd/mimir/help-all.txt.tmpl [hunk 6]
```diff
@@ -1407,6 +1447,64 @@ Usage of ./cmd/mimir/mimir:
...
+  -ingester.partition-ring.etcd.endpoints string
+    	The etcd endpoints to connect to.
+  -ingester.partition-ring.etcd.max-retries int
+    	The maximum number of retries to do for failed ops. (default 10)
+  -ingester.partition-ring.etcd.password string
+    	Etcd password.
+  -ingester.partition-ring.etcd.tls-ca-path string
+    	Path to the CA certificates to validate server certificate against. If not set, the host's root CA certificates are used.
+  -ingester.partition-ring.etcd.tls-cert-path string
+    	Path to the client certificate, which will be used for authenticating with the server. Also requires the key path to be configured.
```
Comment (tacole02) [inline cmd/mimir/help-all.txt.tmpl:1477]:
```suggestion
    	Path to the client certificate, which is used for authenticating with the server. This setting also requires the key path to be configured.
```

File: cmd/mimir/help-all.txt.tmpl [hunk 7]
```diff
@@ -1407,6 +1447,64 @@ Usage of ./cmd/mimir/mimir:
...
+  -ingester.partition-ring.etcd.tls-min-version string
+    	Override the default minimum TLS version. Allowed values: VersionTLS10, VersionTLS11, VersionTLS12, VersionTLS13
+  -ingester.partition-ring.etcd.tls-server-name string
+    	Override the expected name on the server certificate.
+  -ingester.partition-ring.etcd.username string
+    	Etcd username.
+  -ingester.partition-ring.min-partition-owners-count int
+    	Minimum number of owners to wait before a PENDING partition gets switched to ACTIVE. (default 1)
+  -ingester.partition-ring.min-partition-owners-duration duration
+    	How long the minimum number of owners should have been enforced before a PENDING partition gets switched to ACTIVE. (default 10s)
```
Comment (tacole02) [inline cmd/mimir/help-all.txt.tmpl:1495]:
```suggestion
    	How long the minimum number of owners is enforced before a PENDING partition switches to ACTIVE. (default 10s)
```

File: cmd/mimir/help.txt.tmpl [hunk 8]
```diff
@@ -389,6 +389,42 @@ Usage of ./cmd/mimir/mimir:
     	Print basic help.
   -help-all
     	Print help, also including advanced and experimental parameters.
+  -ingest-storage.enabled
+    	True to enable the ingestion via object storage.
+  -ingest-storage.kafka.address string
+    	The Kafka backend address.
+  -ingest-storage.kafka.auto-create-topic-default-partitions int
+    	When auto-creation of Kafka topic is enabled and this value is positive, Kafka's num.partitions configuration option is set on Kafka brokers with this value when Mimir component that uses Kafka starts. This configuration option specifies the default number of partitions that Kafka broker will use for auto-created topics. Note that this is Kafka-cluster wide setting, and applies to any auto-created topic. If setting of num.partitions fails, Mimir will proceed anyway, but auto-created topic may have incorrect number of partitions.
```
Comment (tacole02) [inline cmd/mimir/help.txt.tmpl:397]:
```suggestion
    	When auto-creation of Kafka topic is enabled and this value is positive, Kafka's num.partitions configuration option is set on Kafka brokers with this value when Mimir component that uses Kafka starts. This configuration option specifies the default number of partitions that Kafka broker uses for auto-created topics. Note that this is a Kafka-cluster wide setting, and applies to any auto-created topic. If the setting of num.partitions fails, Mimir proceeds anyway, but auto-created topics could have an incorrect number of partitions.
```

File: cmd/mimir/help.txt.tmpl [hunk 9]
```diff
@@ -389,6 +389,42 @@ Usage of ./cmd/mimir/mimir:
...
+  -ingest-storage.kafka.auto-create-topic-enabled
+    	Enable auto-creation of Kafka topic if it doesn't exist. (default true)
+  -ingest-storage.kafka.client-id string
+    	The Kafka client ID.
+  -ingest-storage.kafka.consume-from-position-at-startup string
+    	From which position to start consuming the partition at startup. Supported options: last-offset, start, end, timestamp. (default "last-offset")
+  -ingest-storage.kafka.consume-from-timestamp-at-startup int
+    	Milliseconds timestamp after which the consumption of the partition starts at startup. Only applies when consume-from-position-at-startup is timestamp
+  -ingest-storage.kafka.consumer-group string
+    	The consumer group used by the consumer to track the last consumed offset. The consumer group must be different for each ingester. If the configured consumer group contains the '<partition>' placeholder, it will be replaced with the actual partition ID owned by the ingester. When empty (recommended), Mimir will use the ingester instance ID to guarantee uniqueness.
```
Comment (tacole02) [inline cmd/mimir/help.txt.tmpl:407]:
```suggestion
    	The consumer group used by the consumer to track the last consumed offset. The consumer group must be different for each ingester. If the configured consumer group contains the '<partition>' placeholder, it is replaced with the actual partition ID owned by the ingester. When empty (recommended), Mimir uses the ingester instance ID to guarantee uniqueness.
```

File: cmd/mimir/help.txt.tmpl [hunk 10]
```diff
@@ -389,6 +389,42 @@ Usage of ./cmd/mimir/mimir:
...
+  -ingest-storage.kafka.dial-timeout duration
+    	The maximum time allowed to open a connection to a Kafka broker. (default 2s)
+  -ingest-storage.kafka.last-produced-offset-poll-interval duration
+    	How frequently to poll the last produced offset, used to enforce strong read consistency. (default 1s)
+  -ingest-storage.kafka.last-produced-offset-retry-timeout duration
+    	How long to retry a failed request to get the last produced offset. (default 10s)
+  -ingest-storage.kafka.max-consumer-lag-at-startup duration
+    	The maximum tolerated lag before a consumer is considered to have caught up reading from a partition at startup, becomes ACTIVE in the hash ring and passes the readiness check. Set 0 to disable waiting for maximum consumer lag being honored at startup. (default 15s)
+  -ingest-storage.kafka.producer-max-record-size-bytes int
+    	The maximum size of a Kafka record data that should be generated by the producer. An incoming write request bigger than this size is split into multiple Kafka records. We strongly recommend to not change this setting unless for testing purposes. (default 15983616)
```
Comment (tacole02) [inline cmd/mimir/help.txt.tmpl:419]:
```suggestion
    	The maximum size of a Kafka record data that should be generated by the producer. An incoming write request larger than this size is split into multiple Kafka records. We strongly recommend to not change this setting unless for testing purposes. (default 15983616)
```

File: cmd/mimir/help.txt.tmpl [hunk 11]
```diff
@@ -389,6 +389,42 @@ Usage of ./cmd/mimir/mimir:
...
+  -ingest-storage.kafka.topic string
+    	The Kafka topic name.
+  -ingest-storage.kafka.write-clients int
+    	The number of Kafka clients used by producers. When the configured number of clients is greater than 1, partitions are sharded among Kafka clients. An higher number of clients may provide higher write throughput at the cost of additional Metadata requests pressure to Kafka. (default 1)
```
Comment (tacole02) [inline cmd/mimir/help.txt.tmpl:423]:
```suggestion
    	The number of Kafka clients used by producers. When the configured number of clients is greater than 1, partitions are sharded among Kafka clients. A higher number of clients could provide higher write throughput at the cost of additional Metadata requests pressure to Kafka. (default 1)
```

File: cmd/mimir/help.txt.tmpl [hunk 12]
```diff
@@ -389,6 +389,42 @@ Usage of ./cmd/mimir/mimir:
...
+  -ingest-storage.kafka.write-timeout duration
+    	How long to wait for an incoming write request to be successfully committed to the Kafka backend. (default 10s)
+  -ingest-storage.migration.distributor-send-to-ingesters-enabled
+    	When both this option and ingest storage is enabled, distributors will both write to Kafka and ingesters. A write request will be considered successful only when written to both backends.
```
Comment (tacole02) [inline cmd/mimir/help.txt.tmpl:427]:
```suggestion
    	When both this option and ingest storage is enabled, distributors write to both Kafka and ingesters. A write request is considered successful only when written to both backends.
```

File: cmd/mimir/help.txt.tmpl [hunk 13]
```diff
@@ -397,6 +433,22 @@ Usage of ./cmd/mimir/mimir:
     	The maximum number of in-memory series per metric name, across the cluster before replication. 0 to disable.
   -ingester.max-global-series-per-user int
     	The maximum number of in-memory series per tenant, across the cluster before replication. 0 to disable. (default 150000)
+  -ingester.partition-ring.consul.hostname string
+    	Hostname and port of Consul. (default "localhost:8500")
+  -ingester.partition-ring.delete-inactive-partition-after duration
+    	How long to wait before an INACTIVE partition is eligible for deletion. The partition will be deleted only if it has been in INACTIVE state for at least the configured duration and it has no owners registered. A value of 0 disables partitions deletion. (default 13h0m0s)
```
Comment (tacole02) [inline cmd/mimir/help.txt.tmpl:439]:
```suggestion
    	How long to wait before an INACTIVE partition is eligible for deletion. The partition is deleted only if it has been in INACTIVE state for at least the configured duration and has no owners registered. A value of 0 disables partitions deletion. (default 13h0m0s)
```

File: docs/sources/mimir/configure/configuration-parameters/index.md [hunk 14]
```diff
@@ -194,6 +194,111 @@ api:
...
+    [dial_timeout: <duration> | default = 2s]
+
+    # How long to wait for an incoming write request to be successfully
+    # committed to the Kafka backend.
+    # CLI flag: -ingest-storage.kafka.write-timeout
+    [write_timeout: <duration> | default = 10s]
+
+    # The number of Kafka clients used by producers. When the configured number
+    # of clients is greater than 1, partitions are sharded among Kafka clients.
+    # An higher number of clients may provide higher write throughput at the
```
Comment (tacole02) [inline docs/sources/mimir/configure/configuration-parameters/index.md:226]:
```suggestion
    # A higher number of clients may provide higher write throughput at the
```

File: docs/sources/mimir/configure/configuration-parameters/index.md [hunk 15]
```diff
@@ -194,6 +194,111 @@ api:
...
+    # cost of additional Metadata requests pressure to Kafka.
+    # CLI flag: -ingest-storage.kafka.write-clients
+    [write_clients: <int> | default = 1]
+
+    # The consumer group used by the consumer to track the last consumed offset.
+    # The consumer group must be different for each ingester. If the configured
+    # consumer group contains the '<partition>' placeholder, it will be replaced
```
Comment (tacole02) [inline docs/sources/mimir/configure/configuration-parameters/index.md:233]:
```suggestion
    # consumer group contains the '<partition>' placeholder, it is replaced
```

File: docs/sources/mimir/configure/configuration-parameters/index.md [hunk 16]
```diff
@@ -194,6 +194,111 @@ api:
...
+    # with the actual partition ID owned by the ingester. When empty
+    # (recommended), Mimir will use the ingester instance ID to guarantee
```
Comment (tacole02) [inline docs/sources/mimir/configure/configuration-parameters/index.md:235]:
```suggestion
    # (recommended), Mimir uses the ingester instance ID to guarantee
```

File: docs/sources/mimir/configure/configuration-parameters/index.md [hunk 17]
```diff
@@ -194,6 +194,111 @@ api:
...
+
+    # Enable auto-creation of Kafka topic if it doesn't exist.
+    # CLI flag: -ingest-storage.kafka.auto-create-topic-enabled
+    [auto_create_topic_enabled: <boolean> | default = true]
+
+    # When auto-creation of Kafka topic is enabled and this value is positive,
+    # Kafka's num.partitions configuration option is set on Kafka brokers with
+    # this value when Mimir component that uses Kafka starts. This configuration
+    # option specifies the default number of partitions that Kafka broker will
+    # use for auto-created topics. Note that this is Kafka-cluster wide setting,
```
Comment (tacole02) [inline docs/sources/mimir/configure/configuration-parameters/index.md:281]:
```suggestion
    # option specifies the default number of partitions that the Kafka broker
    # uses for auto-created topics. Note that this is a Kafka-cluster wide setting,
```

File: docs/sources/mimir/configure/configuration-parameters/index.md [hunk 18]
```diff
@@ -194,6 +194,111 @@ api:
...
+    # and applies to any auto-created topic. If setting of num.partitions fails,
```
Comment (tacole02) [inline docs/sources/mimir/configure/configuration-parameters/index.md:282]:
```suggestion
    # and applies to any auto-created topic. If the setting of num.partitions fails,
```

File: docs/sources/mimir/configure/configuration-parameters/index.md [hunk 19]
```diff
@@ -194,6 +194,111 @@ api:
...
+    # Mimir will proceed anyway, but auto-created topic may have incorrect
+    # number of partitions.
```
Comment (tacole02) [inline docs/sources/mimir/configure/configuration-parameters/index.md:284]:
```suggestion
    # Mimir proceeds anyways, but auto-created topics could have an incorrect
    # number of partitions.
```

File: docs/sources/mimir/configure/configuration-parameters/index.md [hunk 20]
```diff
@@ -194,6 +194,111 @@ api:
...
+    # CLI flag: -ingest-storage.kafka.auto-create-topic-default-partitions
+    [auto_create_topic_default_partitions: <int> | default = 0]
+
+    # The maximum size of a Kafka record data that should be generated by the
+    # producer. An incoming write request bigger than this size is split into
```
Comment (tacole02) [inline docs/sources/mimir/configure/configuration-parameters/index.md:289]:
```suggestion
    # producer. An incoming write request larger than this size is split into
```

File: docs/sources/mimir/configure/configuration-parameters/index.md [hunk 21]
```diff
@@ -194,6 +194,111 @@ api:
...
+    # multiple Kafka records. We strongly recommend to not change this setting
+    # unless for testing purposes.
+    # CLI flag: -ingest-storage.kafka.producer-max-record-size-bytes
+    [producer_max_record_size_bytes: <int> | default = 15983616]
+
+  migration:
+    # When both this option and ingest storage is enabled, distributors will
+    # both write to Kafka and ingesters. A write request will be considered
+    # successful only when written to both backends.
```
Comment (tacole02) [inline docs/sources/mimir/configure/configuration-parameters/index.md:298]:
```suggestion
    # When both this option and ingest storage are enabled, distributors
    # write to both Kafka and ingesters. A write request is considered
    # successful only when written to both backends.
```

File: docs/sources/mimir/configure/configuration-parameters/index.md [hunk 22]
```diff
@@ -1114,6 +1219,64 @@ ring:
   # CLI flag: -ingester.ring.spread-minimizing-zones
   [spread_minimizing_zones: <string> | default = ""]
 
+partition_ring:
+  # The key-value store used to share the hash ring across multiple instances.
+  # This option needs be set on ingesters, distributors, queriers and rulers
```
Comment (tacole02) [inline docs/sources/mimir/configure/configuration-parameters/index.md:1224]:
```suggestion
  # This option needs be set on ingesters, distributors, queriers, and rulers
```

File: docs/sources/mimir/configure/configuration-parameters/index.md [hunk 23]
```diff
@@ -1114,6 +1219,64 @@ ring:
...
+      # (advanced) Timeout for storing value to secondary store.
+      # CLI flag: -ingester.partition-ring.multi.mirror-timeout
+      [mirror_timeout: <duration> | default = 2s]
+
+  # Minimum number of owners to wait before a PENDING partition gets switched to
+  # ACTIVE.
+  # CLI flag: -ingester.partition-ring.min-partition-owners-count
+  [min_partition_owners_count: <int> | default = 1]
+
+  # How long the minimum number of owners should have been enforced before a
```
Comment (tacole02) [inline docs/sources/mimir/configure/configuration-parameters/index.md:1268]:
```suggestion
  # How long the minimum number of owners are enforced before a
```

File: docs/sources/mimir/configure/configuration-parameters/index.md [hunk 24]
```diff
@@ -1114,6 +1219,64 @@ ring:
...
+  # PENDING partition gets switched to ACTIVE.
+  # CLI flag: -ingester.partition-ring.min-partition-owners-duration
+  [min_partition_owners_duration: <duration> | default = 10s]
+
+  # How long to wait before an INACTIVE partition is eligible for deletion. The
+  # partition will be deleted only if it has been in INACTIVE state for at least
```
Comment (tacole02) [inline docs/sources/mimir/configure/configuration-parameters/index.md:1274]:
```suggestion
  # partition is deleted only if it has been in INACTIVE state for at least
```

File: docs/sources/mimir/configure/configuration-parameters/index.md [hunk 25]
```diff
@@ -3594,6 +3759,18 @@ The `limits` block configures default and per-tenant limits imposed by component
 # through OTLP.
 # CLI flag: -distributor.otel-metric-suffixes-enabled
 [otel_metric_suffixes_enabled: <boolean> | default = false]
+
+# (experimental) The default consistency level to enforce to queries when using
```
Comment (tacole02) [inline docs/sources/mimir/configure/configuration-parameters/index.md:3763]:
```suggestion
# (experimental) The default consistency level to enforce for queries when using
```

**Review by tacole02 — APPROVED** [review]

**Review by tacole02 — APPROVED** [review]

**Review by pstibrany — APPROVED** [review]
I'm fine adding it to unreleased section, since all the options are still hidden in 2.13.

I'd suggest documenting the following mixin config options:

* `ingest_storage_enabled` -- enables alerts for ingest-storage
* `show_ingest_storage_panels` -- shows ingest-storage panels

**Comment by pracucci** [comment]
Thanks @tacole02. I should have addressed all your comments.

**Comment by pstibrany** [comment]
> I'm fine adding it to unreleased section, since all the options are still hidden in 2.13.
> 
> I'd suggest documenting the following mixin config options:
> 
> * `ingest_storage_enabled` -- enables alerts for ingest-storage
> * `show_ingest_storage_panels` -- shows ingest-storage panels

I meant documenting them in the changelog. However `show_ingest_storage_panels` is already mentioned in the changelog. `ingest_storage_enabled` comes from #7867, which is missing changelog entry, but alerts that it enables are already documented in

```
* [ENHANCEMENT] Alerts: add the following alerts when using ingest-storage: #7699 #7702
  * `MimirIngesterLastConsumedOffsetCommitFailed`
  * `MimirIngesterFailedToReadRecordsFromKafka`
  * `MimirIngesterKafkaFetchErrorsRateTooHigh`
  * `MimirStartingIngesterKafkaReceiveDelayIncreasing`
  * `MimirRunningIngesterReceiveDelayTooHigh`
  * `MimirIngesterFailsToProcessRecordsFromKafka`
  * `MimirIngesterFailsEnforceStrongConsistencyOnReadPath`
```

entry.

**Comment by pracucci** [comment]
> ingest_storage_enabled comes from https://github.com/grafana/mimir/pull/7867, which is missing changelog entry, but alerts that it enables are already documented in <list> entry.

I retro-actively added the PR number to that entry.

**Pull request merged**